pre_commit/staged_files_only.py

```diff
@@ -57,6 +57,12 @@ def _unstaged_changes_cleared(patch_dir: str) -> Generator[None, None, None]:
             patch_file.write(diff_stdout_binary)
 
         # Clear the working directory of unstaged changes
+        #
+        # Do not use "git checkout" to avoid triggering post-checkout hooks.
+        #
+        #       https://github.com/pre-commit/pre-commit/issues/1418
+        #
+        os.environ['_PRE_COMMIT_SKIP_POST_CHECKOUT'] = '1'
```

Comment:
there's `pre_commit.envcontext` to do this sort of thing -- but probably just need to pass it to the individual `checkout` calls instead of applying to the whole thing

can probably do `env = (**os.environ, _PRE_COMMIT...)`

Reply:
I hadn't seen this, thanks!